File: squiggle-app/squiggle-ui/src/types/play.ts
export interface Position {
  x: number;
  y: number;
}

export interface BallState {
  position: Position;
  attachedTo: {
    type: 'attacking' | 'defensive';
    id: number;
  } | null;
}

export interface PlayerState {
  playerId: string;
  position: Position;
  timestamp: number;
  ballState?: BallState;
}

export interface Play {
  id: string;
  name: string;
  createdAt: string;
  playerStates: PlayerState[];
}

export interface CreatePlayRequest {
  name: string;
  playerStates: PlayerState[];
} 

File: squiggle-app/squiggle-backend/internal/models/play.go
package models

import (
	"time"
)

// Position represents a player's position at a specific time
type Position struct {
	X float64 `json:"x"`
	Y float64 `json:"y"`
}

// BallState represents the ball's state at a specific time
type BallState struct {
	Position   Position `json:"position"`
	AttachedTo *struct {
		Type string `json:"type"`
		ID   int    `json:"id"`
	} `json:"attachedTo"`
}

// PlayerState represents a player's state at a specific time
type PlayerState struct {
	PlayerID  string     `json:"playerId"`
	Position  Position   `json:"position"`
	Timestamp int64      `json:"timestamp"`
	BallState *BallState `json:"ballState,omitempty"`
}

// Play represents a recorded play
type Play struct {
	ID           string        `json:"id"`
	Name         string        `json:"name"`
	CreatedAt    time.Time     `json:"createdAt"`
	PlayerStates []PlayerState `json:"playerStates"`
}

// CreatePlayRequest represents the request to create a new play
type CreatePlayRequest struct {
	Name         string        `json:"name"`
	PlayerStates []PlayerState `json:"playerStates"`
}
